### PR TITLE
Mention syntax ambiguity for eRFC 2947.

### PR DESCRIPTION
**Pull request by davidtwco**
Part of rust-lang/rust#53668.

cc @centril

### CODE REVIEW

File: src/expressions/if-expr.md [hunk 1]
```diff
@@ -111,6 +111,12 @@ match EXPR {
 }
 ```
 
+If the expression is a [lazy boolean operator expression][_LazyBooleanOperatorExpression_] then
+this is a error as it is ambiguous which is problematic for the implementation
+of if-let chains (see [eRFC 2947][_eRFCIfLetChain_]).
```
Comment (Centril) [inline src/expressions/if-expr.md]: Perhaps add a note about wrapping in parenthesis as a solution?

File: src/expressions/if-expr.md [hunk 2]
```diff
@@ -111,6 +111,27 @@ match EXPR {
 }
 ```
 
+If the expression is a [lazy boolean operator expression][_LazyBooleanOperatorExpression_] then
```
Comment (Havvy) [inline src/expressions/if-expr.md]: This is a very roundabout way of saying "The expression cannot be a lazy boolean operator expression."

File: src/expressions/if-expr.md [hunk 3]
```diff
@@ -111,6 +111,27 @@ match EXPR {
...
+this is a error as it is ambiguous which is problematic for the implementation
```
Comment (Havvy) [inline src/expressions/if-expr.md]: I'd leave out the "which is problematic". Just state it's ambiguous with a planned future change in the language.

File: src/expressions/if-expr.md (hunk 2 above)
Comment (davidtwco) [reply src/expressions/if-expr.md]: Changed this.

File: src/expressions/if-expr.md (hunk 3 above)
Comment (davidtwco) [reply src/expressions/if-expr.md]: Changed this.

**Comment by davidtwco** [comment]
Oops, let me remove the `fixup!` commit.

**Comment by Centril** [comment]
Thanks!

**Pull request merged**